### ISSUE TITLE
Change of behaviour in InterventionalRadiologyController

### ISSUE THREAD

**Issue by Camille-K** [sofa-framework/BeamAdapter#87]
I believe that the modifications made in #62 (specifically in [this commit](https://github.com/sofa-framework/BeamAdapter/pull/62/commits/97fdf74e7a3b65ed8eba7a37fc7633d9f9e3810d)) introduced a slight change of behaviour in the `interventionalRadiologyComputeSampling`  method of the *InterventionalRadiologyController*.

I have the impression that these lines (487-493) in the former version :
```
  if (curvAbs_xP>0.0)   // all the noticiable point that have a negative curv abs are not simulated => considered as outside of the patient...
  {
      newCurvAbs.push_back(curvAbs_xP);

      if (curvAbs_xP > maxAbsLength)
          maxAbsLength=curvAbs_xP;
  }
```
implied that the keyPoints of each instrument were always taken into account when computing the new curvilinear abscissas, whether the instrument was visible or not.
Whereas in the new version, I think that these keyPoints are only added to the new curvilinear abscissas if the instrument is visible (i.e. if the tip of the instrument is further deployed than the tips of outter instruments with a higher radius).

NB: the second condition in the lines above, however, seems to be unused in the rest of the method
NB2: it seems I can't assign you on this @epernod, I don't have access to the functionality :upside_down_face:

**Comment by epernod** [comment]
Could you give me the scenario were you have a change of behavior. Use case and maybe a small scene to test.
Indeed the method was very tricky to read with a lot of conditions. I tried to rewrite the method to make it simpler and refactor code but I only tested it on the main scene in the plugin. 
Normally it was really just factorizing code but maybe your use case may fall into a specific condition I couldn't test.

**Comment by Camille-K** [comment]
Unfortunately, my use case won't be very relevant here, as I first observed the change of behaviour in a navigation scene using the Cosserat plugin, with a version of the `InterventionalRadiologyController` that I wrote in Python and adapted to the Cosserat model... But I think the 'sampling curvilinear abscissas' part of my code is similar enough to `interventionalRadiologyComputeSampling` for the problem to be present on both sides.

As an alternative, I tried to launch the *3instruments_collis.scn* scene in `BeamAdapter/examples` to reproduce the issue. It is not a minimal working scenario, but it apparently confirms the change of behaviour.
In the example below, all I'm doing is to start the simulation, which triggers the automatic deployment of a first instrument (0). And ~10s in the simulation, I switch to instrument 1 (Ctrl+1), which stops the progress of Instrument0 and starts the deployment of Instrument1.
Below are two clips of this scene, acquired with two different versions of BeamAdapter's master. The first one was made with a 2-month-old version (commit `792a0b9` on 2022-10-11). The second one was made with the latest version (commit `8266631` on 2022-12-28)

https://user-images.githubusercontent.com/34058933/215540534-0c12712b-9a47-43a1-b696-007899ce3617.mp4


https://user-images.githubusercontent.com/34058933/215540560-6f5c6c4c-82d1-4d33-af2d-b2fc7304dc74.mp4

You can see that the progression of Instrument1 inside Instrument0 is visible in the old version, but is not visible with the new one.

I believe the reason for the difference is that the end point of Instrument1 is now only taken into account if Instrument1 is visible (which was not the case before).

**Comment by damienmarchal** [comment]
Thanks @Camille-K  for reporting regression.

**Comment by epernod** [comment]
This issue has been fixed by PR #88
